api/provider/aws/retry.go

```diff
@@ -0,0 +1,24 @@
+package aws
+
+import (
+	"time"
+
+	"github.com/quintilesims/layer0/common/config"
+	"github.com/quintilesims/layer0/common/errors"
+)
+
+func Retry(id string, describeFN func(id string) error) error {
```

Comment:
I'd expect a generic retry function to have a signature like:
```
func retry(timeout, delay time.Duration, fn func() error) error   
-- or possibly --
func retry(timeout, delay time.Duration, fn func() error) (bool, error)
```

The `id string` pass through in this current impl doesn't make a lot of sense to me, and the lack of ability to change the timeout and delay params seems like this would be difficult to test (unless maybe you used a `timeMultiplier`?). We will also need a clear way to determine when to retry and when to error out (hence the two signatures) above).